### PR TITLE
[AF-1528] 7.11.x Workbench offline 

### PR DESCRIPTION
**Pull request by desmax74**
Workbench offline 7.11.x
https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/829
https://github.com/kiegroup/kie-soup/pull/57
https://github.com/kiegroup/appformer/pull/459
https://github.com/kiegroup/kie-wb-common/pull/2117
https://github.com/kiegroup/kie-wb-distributions/pull/811
@porcelli @domhanak @jhrcek

### CODE REVIEW

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2ServletContextListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    private Artifact getArtifact() {
+        Artifact jarArtifact = new DefaultArtifact("org.uberfire",
+                                                   "uberfire-m2repo-editor-backend",
+                                                   "jar",
+                                                   "100-SNAPSHOT");
+        return jarArtifact;
+    }
+
+    @Test
+    public void deployJarsFrowWar() throws Exception {
```
Comment (porcelli) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2ServletContextListenerTest.java]: please add check for .pom files as well

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+                    Path pomDir = Files.createDirectory(tempDir.resolve(jarFileName));
+                    target = Files.createFile(pomDir.resolve(jarFileName.substring(0, jarFileName.length() - 4) + ".pom"));
+                    InputStream stream = jarFile.getInputStream(entry);
+                    java.nio.file.Files.copy(stream, target, StandardCopyOption.REPLACE_EXISTING);
+                    break;
+                }
+            }
+            pom = new SubArtifact(jarArtifact, null, "pom").setFile(target.toFile());
+        } catch (Exception ex) {
+            pom = null;
```
Comment (porcelli) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java]: please log issues here.

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ModuleRepositoryResolverImpl.java [hunk 3]
```diff
@@ -171,6 +183,31 @@ public ModuleRepositoryResolverImpl(final @Named("ioStrategy") IOService ioServi
         return repositories;
     }
 
+    private RepositorySystemSession newSession(RepositorySystem system) {
+        ArtifactRepositoryPreference artifactRepositoryPreference = new ArtifactRepositoryPreference();
+        artifactRepositoryPreference.defaultValue(artifactRepositoryPreference);
+        DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
+        String global = artifactRepositoryPreference.getGlobalM2RepoDir();
+        if (global == null) {
+            global = "repositories" + File.separator + "kie" + File.separator + "global";
```
Comment (porcelli) [inline uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ModuleRepositoryResolverImpl.java]: not much happy with the `default` global has been spread around. could you please centralize it in just one place?

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java (hunk 2 above)
Comment (desmax74) [reply uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java]: Fixed

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2ServletContextListenerTest.java (hunk 1 above)
Comment (desmax74) [reply uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2ServletContextListenerTest.java]: Fixed on both branches

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ModuleRepositoryResolverImpl.java (hunk 3 above)
Comment (desmax74) [reply uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ModuleRepositoryResolverImpl.java]: Fixed on both branches

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+    private final String JAR_EXT = ".jar";
+    private final String WEB_INF_FOLDER = "WEB-INF";
+    private final String LIB_FOLDER = "lib";
+    private final String JAR_ARTIFACT = "jar";
+    private final String GROUP_ID = "groupId";
+    private final String ARTIFACT_ID = "artifactId";
+    private final String VERSION = "version";
+    private final Path tempDir;
+    private String JARS_FOLDER = File.separator + WEB_INF_FOLDER + File.separator + LIB_FOLDER + File.separator;
```
Comment (porcelli) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java]: `private static final` for constansts, please

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    private String MAVEN_META_INF = "META-INF" + File.separator + "maven";
+
+    public M2ServletContextListener() {
+        Path tempDir = null;
+        try {
+            tempDir = Files.createTempDirectory("pom-extract");
+        } catch (IOException e) {
+        }
```
Comment (porcelli) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java]: log error

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ModuleRepositoryResolverImpl.java [hunk 6]
```diff
@@ -171,6 +182,37 @@ public ModuleRepositoryResolverImpl(final @Named("ioStrategy") IOService ioServi
         return repositories;
     }
 
+    private String getGlobalRepoPath() {
+        ArtifactRepositoryPreference artifactRepositoryPreference = new ArtifactRepositoryPreference();
+        artifactRepositoryPreference.defaultValue(artifactRepositoryPreference);
+        String global = artifactRepositoryPreference.getGlobalM2RepoDir();
+        if (global == null) {
+            global = "repositories" + File.separator + "kie" + File.separator + "global";
```
Comment (porcelli) [inline uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ModuleRepositoryResolverImpl.java]: can't we just reuse an existing default? even better, couldn't we just move this default to the `artifactRepositoryPreference.getGlobalM2RepoDir`? and be able to reuse it also by M2ServletContextListener?

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java (hunk 4 above)
Comment (desmax74) [reply uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java]: fixed on both branches

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java (hunk 5 above)
Comment (desmax74) [reply uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java]: fixed on both branches

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ModuleRepositoryResolverImpl.java (hunk 6 above)
Comment (desmax74) [reply uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ModuleRepositoryResolverImpl.java]: Ok

**Pull request merged**